File: Go/ExampleTests/common/resultvalue_string.go
package common

type ResultValueString struct {
	Sum string
}

func NewResultValueStringFromSlice(v []string) ResultValueString {
	s := ResultValueString{}
	if len(v) > 0 { s.Sum = v[0] }
	return s
}

// NewResultValueStringFromText builds from the text form, e.g. Money as "25 USD".
func NewResultValueStringFromText(text string) ResultValueString {
	parts := RequireTokens(text, 1, "ResultValue")
	return ResultValueString{
		Sum: parts[0],
	}
}

func (s ResultValueString) String() string {
	return Token(s.Sum)
}

func (s ResultValueString) Equals(o ResultValueString) bool {
	return DNCEqual(s.Sum, o.Sum)
}

func EqualResultValueStringSlices(a, b []ResultValueString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/resultvalue_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct ResultValueString {
    pub sum: String,
}

impl ResultValueString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            sum: v.get(0).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 1, "ResultValue");
        Self {
            sum: parts[0].clone(),
        }
    }
}

impl std::fmt::Display for ResultValueString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.sum)].join(" "))
    }
}

impl PartialEq for ResultValueString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.sum, &other.sum)
    }
}
